tests/scripts/local_monitoring_tests/rest_apis/test_import_metadata.py

```diff
@@ -224,3 +224,78 @@ def test_repeated_metadata_import(cluster_type):
+    """
+    Test Description: This test validates the response status code of POST /dsmetadata API by specifying the
+    same datasource name with repeated metadata imports by bringing down prometheus server instance
+    """
+    input_json_file = "../json_files/import_metadata.json"
+    json_data = json.load(open(input_json_file))
+
+    datasource_name = json_data['datasource_name']
+    print("datasource_name = ", datasource_name)
+
+    form_kruize_url(cluster_type)
+
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+    # Import metadata using the specified json
+    response = import_metadata(input_json_file)
+    metadata_json = response.json()
+
+    assert response.status_code == SUCCESS_STATUS_CODE
+
+    # Validate the json against the json schema
+    errorMsg = validate_import_metadata_json(metadata_json, import_metadata_json_schema)
+    assert errorMsg == ""
+
+    json_data = json.load(open(input_json_file))
+    datasource = json_data['datasource_name']
+    # Currently only default cluster_name is supported by kruize
+    cluster_name = "default"
+    response = list_metadata(datasource=datasource, cluster_name=cluster_name)
+
+    list_metadata_json = response.json()
+    assert response.status_code == SUCCESS_200_STATUS_CODE
+
+    # Validate the json values
+    import_metadata_json = read_json_data_from_file(input_json_file)
+    validate_list_metadata_parameters(import_metadata_json, list_metadata_json, cluster_name=cluster_name, namespace="monitoring")
+
+    # Scaling down prometheus deployment and statefulset to zero replicas to bring down prometheus datasource connection
+    scale_deployment("monitoring", "prometheus-operator", 0)
+    scale_deployment("monitoring", "prometheus-adapter", 0)
+    scale_statefulset("monitoring", "prometheus-k8s", 0)
+    time.sleep(10)
+
+    # Repeated Import metadata using the specified json
+    response = import_metadata(input_json_file)
+    metadata_json = response.json()
+
+    assert response.status_code == ERROR_STATUS_CODE
+    assert metadata_json['message'] == IMPORT_METADATA_DATASOURCE_CONNECTION_FAILURE_MSG
+
+    json_data = json.load(open(input_json_file))
+    datasource = json_data['datasource_name']
+    # Currently only default cluster_name is supported by kruize
+    cluster_name = "default"
+    response = list_metadata(datasource=datasource, cluster_name=cluster_name)
+
+    list_metadata_json = response.json()
+    assert response.status_code == SUCCESS_200_STATUS_CODE
+
+    # Validate the json values
+    import_metadata_json = read_json_data_from_file(input_json_file)
+    validate_list_metadata_parameters(import_metadata_json, list_metadata_json, cluster_name=cluster_name, namespace="monitoring")
+
+    #validate namespaces
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+    scale_deployment("monitoring", "prometheus-operator", 1)
```

Comment:
@shreyabiradar07 - Please ensure that once there are no issues when prometheus is enabled back and other tests are executed on both minikube & openshift

Reply:
Local monitoring test results
[PR1211_minikube_local_monitoring_tests.zip](https://github.com/user-attachments/files/15964938/PR1211_minikube_local_monitoring_tests.zip)
[PR1211_openshift_local_monitoring_tests.zip](https://github.com/user-attachments/files/15964942/PR1211_openshift_local_monitoring_tests.zip)

Reply:
Results look fine, can you update the documentation as well

Reply:
Updated `local_monitoring_tests.md `